community/server/src/main/java/org/neo4j/server/rest/discovery/DiscoverableBoltURI.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.server.rest.discovery;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import javax.ws.rs.core.UriInfo;
+
+import org.neo4j.graphdb.config.InvalidSettingException;
+import org.neo4j.helpers.AdvertisedSocketAddress;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.configuration.ConnectorPortRegister;
+import org.neo4j.server.configuration.ServerSettings;
+
+class DiscoverableBoltURI
+{
+    static String calculateDiscoverableBoltURI( Config config, ConnectorPortRegister connectorPortRegister,
```

Comment:
Why not URI as the return type?

Reply:
I started with that, but then realized the value eventually needs to become a string when it's passed to `putAbsoluteUri` in the representation, so I changed it to string to be consistent with managementUri and dataUri both being strings. 

TL;DR: I chose an inferior type because I couldn't deal with three variables all having the same suffix but different types.. I'll change it back :)